### DISCUSSION TITLE
Dokan fs not showing in Disk Manager

### DISCUSSION THREAD

**Discussion by silajim** [dokan-dev/dokany#1012] [Q&A]
In the mirror example there is the Option /O which uses the Windows mount manager, but the "drive" is not visible in the Disk Managment. is this a bug or not implemented? I don't even know if it's possible.
Is there a call that can be done within the userspace in order to do that?  

I have heavily modified the mirror example to meet my needs, but that part never showed the disk in the Disk Manager.

**Comment by Dwedit** [comment]
Registering in the Windows Volume Manager is not necessarily the same thing as showing up in Disk Management.

Here's an example of what happens when a program does _not_ use the Windows Volume Manager: "OSFMOUNT", a program that mounts disk image files.  After your disk image is mounted, you can access files fine, open them, whatever.  But the Windows API call GetFinalPathNameByHandleW will break because the device isn't registered with the Windows Volume Manager.  It will be unable to find the mounted drive letter.

**Reply by silajim** [reply]
I get that, but if someone wanted to, could be done, do you know?

**Reply by Dwedit** [reply]
I'm not sure what the intent would be to have a user-mode filesystem show up in Disk Management.  Disk Management generally deals with physical disks, and the partitions on those disks.  A user mode file system isn't going to be directly involved with partitioning a disk.

It could possibly make sense for a software-defined block device, such as a mounted disk image to appear there, as a block device can contain partitions.

But a user-mode filesystem isn't a block device, and it doesn't have partitions.  I can't think of a good reason for a user-mode filesystem to appear in the Disk Management window.


There are other ways to make drive letters, such as mounting network shares, or using the Subst command.  None of those appear in the Disk Management window, because they're not block devices that contain partitions and filesystems.

**Reply by silajim** [reply]
I know it does not make any sense generally speaking to be seen as a block device. But I want to cover any blocks that Backblaze might add, like instead of getting the volumes, getting the list of physical devices which is why I am looking into this. I am planning to move all the HDD i have into a nas and I want to be sure 100% that I can back them up like now. 

Edit: I am also looking into storage filter drivers and maybe interoperation with dokan. That way someone could map a folder from dokan, so that way the folder acts as the dokan mountpoint.
